docs/software/apptainer.md

```diff
@@ -0,0 +1,233 @@
+     ```
+     module load apptainer/1.1.7-x86_64  squashfuse/0.1.104-x86_64
+     ```
+
+=== "OpenMind"
+
+     Log in a the head node,
+     ```
+     ssh <user>@openmind7.mit.edu
+     ```
+     As a certain amount of computing resources are required to run Singularity, always start with getting an  interactive session on a compute node,
+     ```
+     srun -t 60 --constraint=rocky8 -c 4 --mem=10G --pty bash
+     ```
+     The `constraint=rocky8` is to request a node with the Rocky 8 OS. 
+    
+     Check available Apptainer versions in modules,
+     ```
+     module av openmind8/apptainer
+     ```
+     Load an Apptainer module, for example, 
+     ```
+     module load openmind8/apptainer/1.1.7
+     ```
+
+!!! note 
+    Apptainer modules support both apptainer and singularity commands.
+
+### Download an image
+
+Search for an image that provides your target application, for exmaple on [Docker Hub](https://hub.docker.com/). Here is an example for downloading a Docker image to support Pytorch,
+```
+singularity pull my-image.sif docker://bitnami/pytorch:latest
+```
+The `my-image.sif` is the name of the image. You can name it as you like. 
+
+!!! note 
+    In Apptainer, the command `singularity` is a soft link to an executable named `apptainer`, so all `singularity` commands on this page can be replaced by the `apptainer` command. They work the same. 
+
+
+### Run a program interactively
+
+When the image is ready, launch a container based on the image and then run your application in the container. If you want to work interactively to test and debug codes, it is convineient to log in the containe shell, for exmaple, 
+```
+$ singularity shell my-image.sif 
+Apptainer> python
+Python 3.11.9 (main, May 13 2024, 16:49:42) [GCC 12.2.0] on linux
+Type "help", "copyright", "credits" or "license" for more information.
+>>> import torch
+>>> # Run your programs here.
+```
+
+Alternatively, execute a command in the container to run your programs, 
+```
+singularity exec my-image.sif python my-code.py
+```
+
+Use the full path to the image file if it is not in the current directory. 
+
+The `python` here is installed in the container and has nothing to do with the `python` or `anaconda` modules that have been installed on the host. As the python environment in the container provides all pacakges you need, you don't need to install any python packages and their dependecies. Now you can see the advantage of using a conatainer. 
+
+### Submit a batch job
+
+When the tests are completed, you can submit a batch job to run your program in the background. Here is a typical batch job script (e.g. named `job.sh`).
+
+=== "Engaging"
+     
+     ```
+     #!/bin/bash                      
+     #SBATCH -t 01:30:00                  # walltime = 1 hours and 30 minutes
+     #SBATCH -N 1                         # one node
+     #SBATCH -n 2                         # two CPU cores
+     #SBATCH -p sched_mit_psfc_gpu_r8     # a partition with Rocky 8 nodes
```

Comment:
We should avoid using PI or OU resources in documentation examples. If this example only works on Rocky 8 we can put sched_mit_orcd here for now as the partition. Better would be to use a Centos 7 example until we have public Rocky 8 nodes